sdk/core/azure-core/azure/core/utils/_utils.py

```diff
@@ -87,26 +87,79 @@ def case_insensitive_dict(*args: Any, **kwargs: Any) -> MutableMapping:
     :return: A case-insensitive mutable mapping object.
     :rtype: ~collections.abc.MutableMapping
     """
+    return CaseInsensitiveDict(*args, **kwargs)
+
+class CaseInsensitiveDict(MutableMapping):
+    """
+    NOTE: This implementation is heavily inspired from the case insensitive dictionary from the requests library. Thank you !!
+    Case insensitive dictionary implementation.
+    The keys are expected to be strings and will be stored in lower case.
+    case_insensitive_dict = CaseInsensitiveDict()
+    case_insensitive_dict['Key'] = 'some_value'
+    case_insensitive_dict['key'] == 'some_value' #True
+    """
+    def __init__(self, data = None, **kwargs: Any) -> None:
```

Comment:
`data` should be positional only, right? I know we can't use the syntax yet (because we need to support python < 3.8), but what would you expect the result to be if you call `CaseInsensitiveDict(data="what's up?")`? We can emulate the behavior with a `*args` and make sure we get 0 or one arguments if we want, though...

Reply:
FYI. https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_patch.py#L777

Reply:
I was wanting data to be other MutableMappings that one could pass in and then update would automatically handle that. I was expecting  `my_dict = CaseInsensitiveDict(data="what's up?")` to look like `my_dict["data"]="what's up?"` .  Agree *args will be better here to emulate that, because right now it doesn't handle that scenario.